packages/react-core/src/demos/ComposableMenu.md

```diff
@@ -869,14 +868,259 @@ MenuWithFlyout = () => {
+    let favorites = [];
+
+    menuItems.forEach(item => {
+      if (item.type === MenuGroup) {
+        item.props.children.filter(child => {
+          if (favIds.includes(child.props.itemId)) {
+            favorites.push(child);
+          }
+        });
+      } else {
+        if (favIds.includes(item.props.itemId)) {
+          favorites.push(item);
+        }
+      }
+    });
+
+    return favorites;
+  };
+
+  const filterItems = (items, filteredIds) => {
+    if (filteredIds.length === 1 && filteredIds[0] === '*') {
+      return items;
+    }
+    let filteredCopy = items
+      .map(group => {
+        if (group.type === MenuGroup) {
+          let filteredGroup = React.cloneElement(group, {
+            children: group.props.children.filter(child => {
+              if (filteredIds.includes(child.props.itemId) || child.type === Divider) {
+                return child;
+              }
+            })
+          });
+          if (filteredGroup.props.children.length > 0 && filteredGroup.props.children[0].type !== Divider) {
+            return filteredGroup;
+          }
+        } else {
+          if (filteredIds.includes(group.props.itemId)) {
+            return group;
+          }
+        }
+      })
+      .filter(newGroup => newGroup);
+
+    if (filteredCopy.length > 0) {
+      let lastGroup = filteredCopy.pop();
+      if (lastGroup.type === MenuGroup) {
+        lastGroup = React.cloneElement(lastGroup, {
+          children: lastGroup.props.children.filter(item => item.type !== Divider)
+        });
+      }
+      filteredCopy.push(lastGroup);
+    }
+    return filteredCopy;
+  };
+
+  const onTextChange = textValue => {
+    if (textValue === '') {
+      setFilteredIds(['*']);
+      return;
+    }
+
+    const filteredIds = refFullOptions
+      .filter(item => item.innerText.toLowerCase().includes(textValue.toString().toLowerCase()))
+      .map(item => item.id);
+    setFilteredIds(filteredIds);
+  };
+
+  const onFavorite = (event, itemId, actionId) => {
+    event.stopPropagation();
+    if (actionId === 'fav') {
+      const isFavorite = favorites.includes(itemId);
+      if (isFavorite) {
+        setFavorites(favorites.filter(fav => fav !== itemId));
+      } else {
+        setFavorites([...favorites, itemId]);
+      }
+    }
+  };
+
+  const filteredFavorites = filterItems(createFavorites(favorites), filteredIds);
+  const filteredItems = filterItems(menuItems, filteredIds);
+  if (filteredItems.length === 0) {
+    filteredItems.push(<MenuItem key="no-items">No results found</MenuItem>);
+  }
+
+  const menu = (
+    <Menu ref={menuRef} onActionClick={onFavorite} onSelect={(_ev, itemId) => console.log('selected', itemId)}>
+      <MenuInput>
+        <TextInput
+          aria-label="Filter menu items"
+          iconVariant="search"
+          type="search"
+          onChange={value => onTextChange(value)}
+        />
+      </MenuInput>
+      <Divider />
+      <MenuContent>
+        <MenuList>
```

Comment:
This needs to wrap the `<MenuItem>`s. I'm assuming it would wrap `{filteredFavorites}` a few lines below this?

FWIW `<MenuList>` generates a `<ul>` element. The only valid children of a `<ul>` are `<li>` elements (what the `<MenuItem>` generates here). And the inverse is similar, `<li>` elements can only be a child of `<ul>` or `<ol>`.

Reply:
`filteredFavorites` and `filteredItems` are both collections of `MenuItem` / `MenuGroup` so it should be good.